Portchain/stream-sea-client-js
# Fresh group ID for every subscription in fanout mode

## Description

This PR changes the Kafka group-id behavior for fanout clients.
Before, a StreamSeaClient with fanout=true would generate a random group-id on creation and then reuse it for every subscription.
With this PR, a StreamSeaClient with fanout=true will generate a random group-id for every subscription.

## Motivation
Kafka expects consumers with the same groupId to subscribe to the same topics. This was causing errors when clients with the same stream-sea `clientId` subscribed to multiple streams. This PR fixes the fanout=true case.

## Testing
- Create a stream-sea-client in fanout mode
- Add two subscriptions to the client
- Verify no errors are encountered in the stream-sea server
- Examine stream-sea server INFO messages to verify the two subscriptions have different group IDs (i.e. two random UUIDs)
